### DISCUSSION TITLE
local-path volume space usage

### DISCUSSION THREAD

**Discussion by valerauko** [k3s-io/k3s#4592] [Q&A] [answered]
I'm running a k3s cluster and one of the nodes always has `ImageGCFailed` and `FreeDiskSpaceFailed` in its events. It does say 84% of the local disk is used. The node in question has two pods on it with persistent volumes provisioned with local-path.

I tried to look what takes up so much space using `ncdu -x /`. I've found the two volumes mentioned above at /var/lib/rancher/k3s/storage, one of them is 5.5GiB, the other is 2.8GiB.

My issue is that the /var/lib/kubelet/pods/(uuid)/volume-subpaths folder of one of the pods is huge. It's the pod with the 5.5GiB volume attached to two containers. The problem is that while there should be enough disk space for the volume itself, having it mounted (copied?) three times (once in k3s/storage, twice in volume-subpaths) is too much.

Is this behavior normal? Do volumes mounted in a deployment with volume subpaths get copied multiple times with local-path? I expected it to use symlinks for this... Or is this some leftover from a previous version that isn't actually needed anymore and can be deleted? Or should I just stop mounting using subpaths?

**Comment by brandond** [comment, marked as answer]
I suspect that the `ncdu` command you're running is simply reporting finding volume bind-mounted at multiple paths; I'm not aware of anything that would actually cause it to store multiple copies of your data.

**Reply by valerauko** [reply]
I thought (nc)du's -x flag will tell it to ignore mounts, but apparently bind mounts slip through by having the same device id... Well now I'm back to square one trying to figure out what's taking up space then...
